dream_layer_frontend/src/hooks/useHistoryReducer.ts

```
@@ -0,0 +1,158 @@
+
+export function useHistoryReducer<T>(
+  initialState: T,
+  enableKeyboardShortcuts: boolean = true
+): UseHistoryReducerReturn<T> {
+  const [historyState, dispatch] = useReducer(historyReducer<T>, {
+    past: [],
+    present: initialState,
+    future: []
+  });
+
+  const keyboardHandlerRef = useRef<(event: KeyboardEvent) => void | null>(null);
```

Biome (1.9.4)

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)